packages/generate-object/src/index.ts

```diff
@@ -2,45 +2,69 @@ import type { GenerateTextOptions, GenerateTextResult } from '@xsai/generate-tex
 import type { Infer, InferIn, Schema } from 'xsschema'
 
 import { generateText } from '@xsai/generate-text'
-import { toJSONSchema, validate } from 'xsschema'
+import { toJSONSchema, validate, wrapArray, wrapObject } from 'xsschema'
 
 export interface GenerateObjectOptions<T extends Schema> extends GenerateTextOptions {
   schema: T
   schemaDescription?: string
   schemaName?: string
 }
 
-export interface GenerateObjectResult<T extends Schema> extends Omit<GenerateTextResult, 'text'> {
-  object: Infer<T>
-}
+type GenerateResult<O> = GenerateTextResult & { object: O }
+
+type OptionOutput = 'array' | 'object'
+
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output: 'array' }): Promise<GenerateResult<Array<Infer<T>>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output: 'object' }): Promise<GenerateResult<Infer<T>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T>): Promise<GenerateResult<Infer<T>>>
+export async function generateObject<T extends Schema>(options: GenerateObjectOptions<T> & { output?: OptionOutput }) {
+  const { schema: rawSchema } = options
+
+  let wrappedSchema = rawSchema
+  if (options.output === 'array') {
+    wrappedSchema = await wrapObject(await wrapArray(options.schema), 'elements')
+  }
+
+  const schema = await toJSONSchema(wrappedSchema)
 
-/** @experimental WIP */
-export const generateObject = async <T extends Schema>(options: GenerateObjectOptions<T>): Promise<GenerateObjectResult<T>> =>
-  generateText({
+  return generateText({
     ...options,
     response_format: {
       json_schema: {
         description: options.schemaDescription,
         name: options.schemaName ?? 'json_schema',
-        schema: await toJSONSchema(options.schema),
+        schema,
         strict: true,
       },
       type: 'json_schema',
     },
-    schema: undefined,
-    schemaDescription: undefined,
-    schemaName: undefined,
-  })
-    .then(async ({ finishReason, messages, steps, text, toolCalls, toolResults, usage }) => {
-      const object = await validate(options.schema, JSON.parse(text!) as InferIn<T>)
+    schema: undefined, // Remove schema from options
+    schemaDescription: undefined, // Remove schemaDescription from options
+    schemaName: undefined, // Remove schemaName from options
+  }).then(async ({ finishReason, messages, steps, text, toolCalls, toolResults, usage }) => {
+    const ret: Infer<T> = await validate(wrappedSchema, JSON.parse(text!) as InferIn<T>)
 
+    if (options.output === 'array') {
       return {
         finishReason,
         messages,
-        object,
+        object: (ret as { elements: Array<Infer<T>> }).elements,
         steps,
         toolCalls,
         toolResults,
         usage,
       }
-    })
+    }
+
+    return {
+      finishReason,
+      messages,
+      object: ret,
```

Comment:
should we rename `ret` to `object`?

```suggestion
      object,
```